grype/version/version.go

```diff
@@ -63,6 +64,10 @@ func (v *Version) populate() error {
 		ver, err := newDebVersion(v.Raw)
 		v.rich.debVer = ver
 		return err
+	case GolangFormat:
+		ver, err := newGolangVersion(v.Raw)
+		v.rich.golangVersion = ver
+		return err
```

Comment:
Taking a look here - if we do get the error for when something is like `(devel)` then there are 3 references to `NewVersionFromPkg` where the error could be returned. I think the only instance for the case of code changes here would be under `grype/search/language.go` - This is eventually surfaced as a warning so we don't exit grype based on an invalid package version.

I think the warning is good for now since it's how we've handled this kind of error case in the past.

https://github.com/anchore/grype/blob/83e5176127d0d654d9d671504e30d89fa5fd1070/grype/search/criteria.go#L34C1-L38C5

Reply:
It looks like the devel version is already handled specially by the Go Matcher:

https://github.com/anchore/grype/blob/9750ef245281262b2d606967fd3ce665117b7c0c/grype/matcher/golang/matcher.go#L45-L52

And manual testing indicates that we don't get this error logged as a warning.

It might be worth consolidating our golang version handling in the future, but I think for now at least this won't spam a warning for every go executable.